File: MIA/lista10/a.py
n, d = map(int, input().split())
players = sorted(list(map(int, input().split())), reverse=True)
i = 0
j = len(players) - 1
wins = 0
while i <= j:
    # print(f"player {i}, power {players[i]}")
    rest = d // players[i]
    # print(f"needs {rest} players")
    if j - i < rest:
        break
    i += 1
    j -= rest
    # if rest != 0:
    #     players = players[1:-rest]
    # else:
    #     players = players[1:]
    wins += 1
    # print(players[i : j + 1])

print(wins)

File: MIA/lista10/a.cpp
#include <algorithm>
#include <iostream>
#include <vector>
using namespace std;

int main() {
  int n, d;
  cin >> n >> d;
  vector<int> players(n);
  for (int i = 0; i < n; ++i) {
    cin >> players[i];
  }
  sort(players.rbegin(), players.rend());
  int wins = 0;
  while (!players.empty()) {
    int rest = d / players[0];
    if ((int)players.size() < rest + 1) {
      break;
    }
    if (rest != 0) {
      players.erase(players.begin());
      players.erase(players.end() - rest, players.end());
    } else {
      players.erase(players.begin());
    }
    wins++;
  }
  cout << wins << endl;
  return 0;
}
